### PR TITLE
docs: update README.md

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Updated readme

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -44,27 +80,56 @@ import (
...
 
     // Flushes any queued messages and closes the client.
+    
     client.Close()
 }
 ```
 
+## Gzip support
+
+The Go SDK supports Gzip compression from version 4.0.0 and it is set to true by default. However, you can manually enable or disable this feature by setting the `Gzip` parameter to `1` or `0` while initializing the SDK, as shown:
```
Comment (ameypv-rudder) [inline Readme.md:111]: In the [spec](https://www.notion.so/rudderstacks/Update-Go-SDK-readme-5269984801a345b7849526bc367c53cc), we say the following:
"Gzip is set to true by default" and in the snippet, we have `Gzip: 1`.

However, we say "set 1 to disable and 0 to enable", which is contradictory. Could you please confirm the values, @pallabmaiti

Comment (pallabmaiti) [reply Readme.md:111]: Spec updated.

**Review by pallabmaiti — APPROVED** [review]

File: Readme.md [hunk 2]
```diff
@@ -43,27 +80,56 @@ import (
...
+    client, _ := analytics.NewWithConfig(WRITE_KEY,
 		analytics.Config{
-			Interval:  30 * time.Second,
-			BatchSize: 100,
-			Verbose:   true,
+			DataPlaneUrl: DATA_PLANE_URL,
+			Interval:     30 * time.Second,
+			BatchSize:    100,
+			Verbose:      true,
+			Gzip:         1,
```
Comment (ameypv-rudder) [inline Readme.md:93]:
```suggestion
			Gzip:         1,  // Disables Gzip compression - set to 0 to enable Gzip.
```

File: Readme.md [hunk 3]
```diff
@@ -43,27 +80,56 @@ import (
...
 
     // Flushes any queued messages and closes the client.
+    
     client.Close()
 }
 ```
 
+## Gzip support
+
+The Go SDK supports Gzip compression from version 4.0.0 and it is set to true by default. However, you can manually enable or disable this feature by setting the `Gzip` parameter to `1` or `0` while initializing the SDK, as shown:
```
Comment (ameypv-rudder) [inline Readme.md:111]:
```suggestion
The Go SDK supports Gzip compression from version 4.0.0 and it is enabled (set to `0`) by default. However, you can disable this feature by setting the `Gzip` parameter to `1` while initializing the SDK, as shown:
```

Comment (ameypv-rudder) [inline Readme.md:111]:
```suggestion
The Go SDK supports Gzip compression from version 4.0.0 and it is enabled (set to `0`) by default. However, you can disable this feature by setting the `Gzip` parameter to `1` while initializing the SDK, as shown:
```

File: Readme.md [hunk 4]
```diff
@@ -43,27 +80,56 @@ import (
...
+
+```go
+client, _ := analytics.NewWithConfig(WRITE_KEY,
+		analytics.Config{
+			DataPlaneUrl: DATA_PLANE_URL,
+			Interval:     30 * time.Second,
+			BatchSize:    100,
+			Verbose:      true,
+			Gzip:         1  // Set to 0 to disable Gzip compression
```
Comment (ameypv-rudder) [inline Readme.md:120]:
```suggestion
			Gzip:         1  // Set to 0 to enable Gzip compression
```

**Pull request merged**